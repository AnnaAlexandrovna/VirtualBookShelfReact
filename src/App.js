import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
import Home from './containers/Home';
import Subject from './containers/Subjects';
import Favorites from './containers/Favorites';
import { Routers } from './constants/routes';
import Book from './containers/Book';
import Collection from './containers/Collection';
import Search from './containers/Search';
import { LanguageProvider } from './utils/providers/languageProvider';
import { Provider } from 'react-redux';
import { configureStoreVB } from './store';
import { ViewportProvider } from './utils/providers/viewportProvider';
import { Authorization } from './containers/Authorization';
import Login from './containers/Login';
import AddFavorites from './utils/favorites/addFavorites';
import Logout from './containers/Logout';
import Alert from './containers/Alert';

const store = configureStoreVB();
const App = () => {
    return <ViewportProvider>
        <Provider store={store}>
            <AddFavorites />
            <LanguageProvider>
                <Alert>
                    <Router>
                        <Switch>
                            <Route path={Routers.subject} component={Subject} id={Routers.subject} />
                            <Route path={Routers.favorites} component={Favorites} />
                            <Route path={Routers.book} component={Book} />
                            <Route path={Routers.collection} component={Collection} />
                            <Route path={Routers.search} component={Search} />
                            <Route path={Routers.authorization} component={Authorization} />
                            <Route path={Routers.login} component={Login} />
                            <Route path={Routers.logout} component={Logout} />
                            <Route path={Routers.home} component={Home} />
                        </Switch>
                    </Router>
                </Alert>
            </LanguageProvider>
        </Provider>
    </ViewportProvider>;
};

export default App;