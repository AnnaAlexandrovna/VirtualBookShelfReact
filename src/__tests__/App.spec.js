import App from '../App';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

let wrapper;
const createComponent = () => shallow(< App />);

describe('Render App', () => {
    beforeAll(() => wrapper = createComponent());
    it('should contain ViewportProvider', () => {
        expect(wrapper.name()).toBe('ViewportProvider');
    });

    it('should contain Provider', () => {
        expect(wrapper.childAt(0).name()).toBe('Provider');
    });

    it('should have Provider with store', () => {
        expect(wrapper.childAt(0).prop('store') === null).toBeFalsy();
        expect(wrapper.childAt(0).prop('store') === undefined).toBeFalsy();
    });

    it('should have AddFavorites', () => {
        expect(wrapper.find('AddFavorites').length).toBe(1);
    });

    it('should have Alert', () => {
        expect(wrapper.find('Alert').length).toBe(1);
    });

    it('should have BrowserRouter', () => {
        expect(wrapper.find('BrowserRouter').length).toBe(1);
    });

    it('should have Switch', () => {
        expect(wrapper.find('Switch').length).toBe(1);
    });
    it('should have 9 Route', () => {
        expect(wrapper.find('Route').length).toBe(9);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});