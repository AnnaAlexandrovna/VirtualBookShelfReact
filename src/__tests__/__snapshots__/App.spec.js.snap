// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App should match snapshot 1`] = `
<ViewportProvider>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <AddFavorites />
    <LanguageProvider>
      <Alert>
        <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              id="/subject"
              path="/subject"
            />
            <Route
              component={[Function]}
              path="/favorites"
            />
            <Route
              component={[Function]}
              path="/book/:path/:bookId"
            />
            <Route
              component={[Function]}
              path="/collection/:collectionId"
            />
            <Route
              component={[Function]}
              path="/search"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              path="/authorization"
            />
            <Route
              component={[Function]}
              path="/login"
            />
            <Route
              component={[Function]}
              path="/logout"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              path="/"
            />
          </Switch>
        </BrowserRouter>
      </Alert>
    </LanguageProvider>
  </Provider>
</ViewportProvider>
`;
